### DISCUSSION TITLE
404 when editing a status?

### DISCUSSION THREAD

**Discussion by solarslurpi** [mastodon/mastodon#23363] [Q&A]
This link: [https://vintageracing.social/api/v1/statuses/109791576089151625](https://vintageracing.social/api/v1/statuses/109791576089151625)   
Goes to a page containing this json:
```
{"id":"109791576089151625","created_at":"2023-02-01T21:21:40.384Z","in_reply_to_id":null,"in_reply_to_account_id":null,"sensitive":false,"spoiler_text":"","visibility":"public","language":"en","uri":"https://vintageracing.social/users/sparrow/statuses/109791576089151625","url":"https://vintageracing.social/@sparrow/109791576089151625","replies_count":0,"reblogs_count":0,"favourites_count":0,"edited_at":null,"favourited":false,"reblogged":false,"muted":false,"bookmarked":false,"pinned":false,"content":"\u003cp\u003eAnother toot.\u003c/p\u003e","filtered":[],"reblog":null,"application":{"name":"Web","website":null},"account":{"id":"109790764507180500","username":"sparrow","acct":"sparrow","display_name":"","locked":false,"bot":false,"discoverable":null,"group":false,"created_at":"2023-02-01T00:00:00.000Z","note":"","url":"https://vintageracing.social/@sparrow","avatar":"https://vintageracing.social/avatars/original/missing.png","avatar_static":"https://vintageracing.social/avatars/original/missing.png","header":"https://vintageracing.social/headers/original/missing.png","header_static":"https://vintageracing.social/headers/original/missing.png","followers_count":0,"following_count":0,"statuses_count":2,"last_status_at":"2023-02-01","noindex":false,"emojis":[],"fields":[]},"media_attachments":[],"mentions":[],"tags":[],"emojis":[],"card":null,"poll":null}
```
I expected to get a 200 but got a 404 instead with this Curl.  What am I missing? (note: I gave the app w/ the associated OAuth admin r/w access).

Thank you.
```
curl -X PUT \
  https://vintageracing.social/api/v1/statuses/109791576089151625 \
  -H 'Authorization: Bearer <secret for OAuth>' \
  -H 'Content-Type: application/json' \
  -d '{
  "status": "new status text",
  "spoiler_text": "optional spoiler text"
}'
```

**Comment by trwnh** [comment]
did you give it regular r/w access? `admin:read` and `admin:write` are completely separate scopes from `read` and `write`. specifically, the method you called only accepts `write` or `write:statuses`.

**Reply by solarslurpi** [reply]
Thank you.  Yes, I misunderstood `admin:read`, `admin:write` to mean something different (then I actually read the doc... :-( ...  

My confusion was/is if it is possible to assign an OAuth to the Mastodon instance (versus a users of the Mastodon instance)?  What I have found to do is only Oauth app per user.  On the instance of Mastodon I run, I want OAuth to work across all users?  Thank you.

**Reply by trwnh** [reply]
@solarslurpi correct me if i'm wrong, but it seems like you want admins to be able to edit user statuses? if so, then there will have to be a new admin api method added for this.

**Comment by solarslurpi** [comment]
Thank you.  Yes.  I'm exploring what options there are for content management.  In this scenario, the owner of the Mastodon instance has installed (for discussion, doesn't exist) the "DignityIndex" package.  This is a python package that background scans the toots on the server and uses "spiffy wonderful algorithms" to determine a Dignity Index.  If the Dignity Index assigned to a toot is below an acceptable threshold, On status_update, the status field still contains the Toot.  However, the spoiler_text is altered to something like 'WARNING - This Toot lacks dignity.' (or something similar).  The thought process I had was this would be a non-destructive first step in content moderation.  The next step is for the admin (person) to go through each of these toots and follow up on administrative action.  Caveat:  I am exploring Mastodon.  I am quite naive and pretty slow so I am trying to figure stuff out while at the same time learn from those who have been there before (particularly around content moderation - get their thoughts...just figuring stuff out right now).
